bridgerelayer/chain/substrate/writer.go

```diff
@@ -0,0 +1,118 @@
+package substrate
+
+import (
+	"context"
+	"encoding/hex"
+
+	"github.com/sirupsen/logrus"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/snowfork/go-substrate-rpc-client/types"
+	"github.com/snowfork/polkadot-ethereum/bridgerelayer/chain"
+)
+
+type Writer struct {
+	conn     *Connection
+	messages <-chan chain.Message
+	log      *logrus.Entry
+}
+
+func NewWriter(conn *Connection, messages <-chan chain.Message, log *logrus.Entry) (*Writer, error) {
+	return &Writer{
+		conn:     conn,
+		messages: messages,
+		log:      log,
+	}, nil
+}
+
+func (wr *Writer) Start(ctx context.Context, eg *errgroup.Group) error {
+	eg.Go(func() error {
+		return wr.writeLoop(ctx)
+	})
+	return nil
+}
+
+func (wr *Writer) writeLoop(ctx context.Context) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case msg := <-wr.messages:
+			err := wr.Write(&msg)
+			if err != nil {
+				wr.log.WithFields(logrus.Fields{
+					"appid": hex.EncodeToString(msg.AppID[:]),
+					"error": err,
+				}).Error("Failure submitting message to substrate")
+			}
+		}
+	}
+}
+
+// SubmitPacket submits a packet, it returns true
+func (wr *Writer) Write(msg *chain.Message) error {
+
```

Comment:
Opinion: standardize function newlines. Some methods do not have a newline and some do.

Reply:
Fixed in latest upload.